File: pola8.c
#include <stdio.h>

int main()
{
    int n;
    scanf("%d", &n);

    // Jika n adalah genap, ubah n ke bentuk ganjil di bawahnya
    if (n % 2 == 0)
        n -= 1;

    // Aku buatnya dengan konsep setengah...jadi setengah atas dan setengah bawah
    int half = n / 2;

    for (int i = 0; i < half; i++)
    {
        for (int j = 0; j < i; j++)
        {
            printf("  ");
        }

        for (int j = i; j < half; j++)
        {
            printf("* ");
        }

        for (int j = i; j <= half; j++)
        {
            printf("* ");
        }
        printf("\n");
    }

    for (int i = 1; i <= half + 1; i++)
    {
        for (int j = i; j <= half; j++)
            printf("  ");

        for (int j = 1; j < i; j++)
        {
            printf("* ");
        }

        for (int j = 1; j <= i; j++)
        {
            printf("* ");
        }
        printf("\n");
    }

    return 0;
}

File: Golang Pattern/Pola8.go
package main

import "fmt"

var length int

func atas (){
    for i:= 0; i < length/2 + 1; i++{
        for j:= 0; j < i ; j++{
            fmt.Print("  ")
        }
        for j:= i; j < length-i; j++{
            fmt.Print("* ")
        }
        for j:= length - i; j < length + 1; j++{
            fmt.Print("  ")
        }
        fmt.Println("")
    }
}

func bawah (){
    for i:= 1; i < length/2 + 1; i++{
        for j:= 0; j < length/2 - i; j++{
            fmt.Print("  ")
        } 
        for j:= length/2 - i; j <= length/2 + i; j++{
            fmt.Print("* ")
        }
        for j:= length/2 + i; j < length; j++{
            fmt.Print("  ")
        }
        fmt.Println("")
    }
}

func main(){
    fmt.Scan(&length)
	if length%2 == 0 {
		length--
	}

    atas()
    bawah()
}
